in-progress/7482-sequencer-prover-test-net.md

```diff
@@ -0,0 +1,409 @@
+
+#### Overview
+
+It is the responsibility of the prover selected at the beginning of an epoch to submit the proof of the epoch.
+
+The proof of epoch i must land in epoch i+1.
+
+```mermaid
+sequenceDiagram
+    participant P as L2 Proposer
+    participant MP as L2 P2P
+    participant PN as L2 Prover Node
+    participant L1R as L1 Rollup Contracts
+
+    PN->>PN: startEpoch
+    P->>MP: Propose TxObjects and L2 block header for a slot 
+    MP->>PN: Broadcast proposal
+    PN->>PN: Execute corresponding TxObjects, cache ProofRequests
+    PN->>L1R: Watches for new L2PendingBlockProcessed events
+    PN->>L1R: Download TxEffects
+    PN->>PN: Submit corresponding ProofRequests for pending block
+    Note over PN: Cache root rollup proof for block
+    Note over PN: After epoch
+    PN->>PN: finalizeEpoch
+    PN->>L1R: Submit proof of epoch to advance Proven Chain
+    L1R->>L1R: Verify blocks in Pending Chain, verify proof
+    L1R->>L1R: Add L2 block to Proven Chain, emit L2ProvenBlockProcessed
+```
+
+#### Confirmation Rules
+
+A node may submit a transaction to the rollup contract on L1 to add blocks from the Pending Chain to the Proven Chain.
+
+The rollup contract will verify that:
+
+- The transaction is submitted by the correct proposer.
+- The specified blocks are in the Pending Chain
+- A proof of the block's correctness has been verified on L1.
+
+After this, all blocks in the epoch are confirmed in the Proven Chain.
+
+### SPRTN v1.0.0 Governance
+
+The SPRTN will have its own governance contract.
+
+Version 1.0.0 will specify an L1 account owner by Aztec Labs that is able to add or remove sequencers and provers. 
+
+
+## Interface
+
+Who are your users, and how do they interact with this? What is the top-level interface?
+
+### Smart Contracts
+The contracts to support this milestone will provide minimal functionality, an overview of their arrangement is shown below.
+![Alt text](./images/7482-contracts.png)
+
+#### Teocalli (Multisig)
+The Teocalli (Aztec great temple) contract will serve as governance for the testnet, for now it is sufficient that it be operated by Aztec Labs. In which case
+its interface will follow that of an arbitrary multisig (gnosis safe will likely be used in practice).
+```sol
+interface ITeocalli {
+  // Execute multisig operations
+  function execute(bytes calldata cmd) external;
+}
+```
+
+#### Validator Registry
+A simple validator registry owned by Teocalli. The addition and removal of validators / sequencers will be permissioned, governed by Teocalli. 
+
+```sol
+interface IValidatorRegistry {
+
+  // Add a validator (who has previously staked) to the staking set.
+  // @dev only admin
+  function addValidator(address validator) external;
+
+  // Remove a currently active validator from the staking set.
+  // @dev only admin
+  function removeValidator(address validator) external;
+
+  // Return the currently active sequencer
+  function getCurrentSequencer() view external returns (address);
```

Comment:
I'm not sure it should be validator's responsibility to determine who the current sequencer/prover is. Anyway, it's easy to move around once we have more clarity.